src/compiler/typescript_to_rust/azle_generate/src/candid_type/errors/wrong_enclosed_type.rs

```diff
@@ -20,19 +20,20 @@ pub struct WrongEnclosedType {
 enum EnclosingType {
     Variant,
     Func,
+    Record,
+    Tuple,
 }
 
 impl WrongEnclosedType {
     pub fn error_from_ts_type_ref(sm_ts_type_ref: &SourceMapped<TsTypeRef>) -> Error {
         let original_location = sm_ts_type_ref.get_location();
-        let name = match sm_ts_type_ref.get_name() {
-            Ok(name) => name.to_string(),
-            Err(err) => return err,
-        };
+        let name = sm_ts_type_ref.get_name();
         let name = match name.as_str() {
             _ if sm_ts_type_ref.symbol_table.variant.contains(&name) => EnclosingType::Variant,
```

Comment:
It's because all of these other types have strict requirements as to what can be the enclosed type. Variant and Record have to be TsTypeLiterals, Func has to be a very specific form of a TsFnOfConstructorType, and tuples need to be a TsTupleType, Opt and Vec can be whatever TsType they want.